ext/mysqli/mysqli.c

```diff
@@ -428,12 +428,20 @@ static int mysqli_object_has_property(zval *object, zval *member, int has_set_ex
 	return ret;
 } /* }}} */
 
-HashTable *mysqli_object_get_debug_info(zval *object, int *is_temp)
+static HashTable *mysqli_object_get_properties_for(zval *object, zend_prop_purpose purpose) /* {{{ */
 {
 	mysqli_object *obj = Z_MYSQLI_P(object);
 	HashTable *retval, *props = obj->prop_handler;
 	mysqli_prop_handler *entry;
 
+	switch (purpose) {
+		case ZEND_PROP_PURPOSE_DEBUG:
+		case ZEND_PROP_PURPOSE_ARRAY_CAST:
+			break;
+		default:
+			return zend_std_get_properties_for(object, purpose);
+	}
```

Comment:
To be clear, what I had in mind is implementing *only* `__getDebugInfo` and dropping the internal `get_debug_info`/`get_properties_for` handler